demo/terraform/main.tf

```diff
@@ -0,0 +1,224 @@
+terraform {
+  required_providers {
+    docker = {
+      source  = "kreuzwerker/docker"
+      version = "~> 2.0"
+    }
+    vault = {
+      source  = "hashicorp/vault"
+      version = "~> 4.0"
+    }
+    local = {
+      source  = "hashicorp/local"
+      version = "~> 2.0"
+    }
+  }
+}
+
+variable "vault_address" {
+  default = "http://127.0.0.1:8200"
+}
+
+provider "docker" {
+  host = "unix:///var/run/docker.sock"  
+}
+
+provider "vault" {
+  address = var.vault_address
+  token   = "myroot"  
+}
+
+resource "docker_image" "vault" {
+  name         = "hashicorp/vault:1.17.5" 
+  keep_locally = false  # Clean up the image if it's no longer used.
+}
+
+resource "docker_container" "vault" {
```

Comment:
We will need to add the container to the kubernetes cluster network. Once we use k3d from PR #3 , the name of the network will be `k3d-gateway-demo`.

Reply:
cool. then I would suggest that you merge first. Otherwise, main will be broken if I add it